common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/CurrentTaskAuthorizationFragment.java

```diff
@@ -0,0 +1,187 @@
+        @Override
+        public void onReceive(Context context, Intent intent) {
+            cancelAuthorization(intent.getBooleanExtra(REQUEST_CANCELLED_BY_USER, false));
+        }
+    };
+
+    void setInstanceState(@NonNull final Bundle instanceStateBundle) {
+        mInstanceState = instanceStateBundle;
+    }
+
+    @SuppressFBWarnings(FindBugsConstants.NP_NULL_ON_SOME_PATH_FROM_RETURN_VALUE)
+    @Override
+    public void onCreate(@Nullable Bundle savedInstanceState) {
+        final String methodName = "#onCreate";
+        super.onCreate(savedInstanceState);
+
+        // Register Broadcast receiver to cancel the auth request
+        // if another incoming request is launched by the app
+        LocalBroadcastManager.getInstance(getContext()).registerReceiver(mCancelRequestReceiver,
+                new IntentFilter(CANCEL_INTERACTIVE_REQUEST));
+
+        if (savedInstanceState == null) {
+            Logger.verbose(TAG + methodName, "Extract state from the intent bundle.");
+            extractState(mInstanceState);
+        } else {
+            // If activity is killed by the os, savedInstance will be the saved bundle.
+            Logger.verbose(TAG + methodName, "Extract state from the saved bundle.");
+            extractState(savedInstanceState);
+        }
+    }
+
+    @SuppressFBWarnings(FindBugsConstants.NP_NULL_ON_SOME_PATH_FROM_RETURN_VALUE)
+    void finish() {
+        final FragmentActivity activity = getActivity();
+        if (activity instanceof AuthorizationActivity) {
+            activity.finish();
+        } else {
+            // The calling activity is not owned by MSAL/Broker.
+            // Just remove this fragment.
+            if (getFragmentManager() != null) {
+                getFragmentManager()
+                        .beginTransaction()
+                        .setTransitionStyle(FragmentTransaction.TRANSIT_FRAGMENT_FADE)
+                        .remove(this)
+                        .commit();
+            }
+        }
+    }
+
+    void extractState(@NonNull final Bundle state) {
+        setDiagnosticContextForNewThread(state.getString(DiagnosticContext.CORRELATION_ID));
+    }
+
+    /**
+     * When authorization fragment is launched.  It will be launched on a new thread. (TODO: verify this)
+     * Initialize based on value provided in intent.
+     */
+    private static String setDiagnosticContextForNewThread(String correlationId) {
+        final String methodName = ":setDiagnosticContextForAuthorizationActivity";
+        final com.microsoft.identity.common.internal.logging.RequestContext rc =
+                new com.microsoft.identity.common.internal.logging.RequestContext();
+        rc.put(DiagnosticContext.CORRELATION_ID, correlationId);
+        DiagnosticContext.setRequestContext(rc);
+        Logger.verbose(
+                TAG + methodName,
+                "Initializing diagnostic context for AuthorizationActivity"
```

Comment:
```suggestion
                "Initializing diagnostic context for CurrentTaskAuthorizationFragment"
```

Reply:
I changed this to CurrentTaskAuthorizationActivity... this code being here is odd, but again, not trying to over correct things here.